File: dotnet/src/ArgonGuard.Passwords/Phc/PhcHash.cs
namespace ArgonGuard.Passwords.Phc;

/// <summary>嚴格解析後的 PHC 欄位。<c>Algorithm</c> 保留原字串供 dispatch 判斷是否 argon2id。</summary>
internal sealed class PhcHash
{
    public PhcHash(string algorithm, long? version, long m, long t, long p,
                   byte[] salt, byte[] tag, bool hasKeyid, bool hasData)
    {
        Algorithm = algorithm;
        Version = version;
        M = m;
        T = t;
        P = p;
        Salt = salt;
        Tag = tag;
        HasKeyid = hasKeyid;
        HasData = hasData;
    }

    public string Algorithm { get; }
    public long? Version { get; }
    public long M { get; }
    public long T { get; }
    public long P { get; }
    public byte[] Salt { get; }
    public byte[] Tag { get; }
    public bool HasKeyid { get; }
    public bool HasData { get; }
}

File: php/src/Phc/PhcHash.php
<?php

declare(strict_types=1);

namespace ArgonGuard\Passwords\Phc;

/**
 * 嚴格解析後的 PHC argon2id 雜湊（salt/tag 為 raw bytes；PHP string 即 byte string）。
 *
 * @internal 不屬於公開 API。
 */
final readonly class PhcHash
{
    public function __construct(
        public string $algorithm,
        public ?int $version,
        public int $m,
        public int $t,
        public int $p,
        public string $salt,
        public string $tag,
        public bool $hasKeyid,
        public bool $hasData,
    ) {
    }
}
